model/project_parser.go

```diff
@@ -589,7 +588,28 @@ func LoadProjectInto(ctx context.Context, data []byte, opts *GetProjectOpts, ide
 				opts.ReadFileFrom = ReadFromPatch
 			}
 		}
-		yaml, err := retrieveFile(ctx, *opts)
+		var yaml []byte
+		if path.Module != "" {
+			module, err := GetModuleByName(intermediateProject.Modules, path.Module)
```

Comment:
you can still have a retrieveFileForModule that itself calls retrieveFile so that you can move this stuff out and make the function more readable.